### ISSUE TITLE
A11Y: Implement navigational (focus) shortcuts

### ISSUE THREAD

**Issue by barklund** [GoogleForCreators/web-stories-wp#4241]
## Feature Description

We start with 3 simple shortcuts for moving between these three sections:

1. <kbd>cmd/win</kbd>+<kbd>opt/alt</kbd>+<kbd>1</kbd> Currently active library tab
1. <kbd>cmd/win</kbd>+<kbd>opt/alt</kbd>+<kbd>2</kbd> Currently selected element(s)
1. <kbd>cmd/win</kbd>+<kbd>opt/alt</kbd>+<kbd>3</kbd> Currently active inspector tab

## Alternatives Considered

* <kbd>cmd</kbd>+<kbd><</kbd>/<kbd>></kbd> for cycling between the three sections, but no clear notation is possible due to different keyboard layouts.

## Additional Context

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

**Comment by o-fernandez** [comment]
@pbakaus a few questions: 

1. Do you have any recommendations or standards on what shortcuts to use? Avoiding clashes with Chrome or [our existing shortcuts](https://docs.google.com/spreadsheets/d/1VNv8O5SnlqZfgYXo5ua2mCc-MWTN9ow6OtYWxml5Ka0/edit#gid=0)?
2. Do you know if implementing these shortcuts is a requirements (A11y-wise) for 1.0 Stable? If so, we'd need to prioritize the scope right away. 
3. Lastly, do we need a "display list" in order to show users the navigational shortcuts? @samitron7 suggested doing that and keeping it separate from the long list of functional keyboard shortcuts (#359)

**Comment by pbakaus** [comment]
> @pbakaus a few questions:
> 
> 1. Do you have any recommendations or standards on what shortcuts to use? Avoiding clashes with Chrome or [our existing shortcuts](https://docs.google.com/spreadsheets/d/1VNv8O5SnlqZfgYXo5ua2mCc-MWTN9ow6OtYWxml5Ka0/edit#gid=0)?

No idea. This is a fantastic question for our internal a11y team.

> 2. Do you know if implementing these shortcuts is a requirements (A11y-wise) for 1.0 Stable? If so, we'd need to prioritize the scope right away.

It is a requirement if there's no other way to complete a CUJ. If there is another way that's not too cumbersome, it is not a requirement.

> 3. Lastly, do we need a "display list" in order to show users the navigational shortcuts? @samitron7 suggested doing that and keeping it separate from the long list of functional keyboard shortcuts (#359)

I would put them where the other functional shortcuts are documented. There's not a huge difference, one is about jumping to a certain point, another one is executing a function.

**Comment by barklund** [comment]
> > 2. Do you know if implementing these shortcuts is a requirements (A11y-wise) for 1.0 Stable? If so, we'd need to prioritize the scope right away.
> 
> It is a requirement if there's no other way to complete a CUJ. If there is another way that's not too cumbersome, it is not a requirement.

**Shortcut to design panel:**
You **_cannot_** complete all CUJ's without the shortcut to the design panel. If you select the non-topmost element on the canvas and press <kbd>tab</kbd> once, the next element up will be focused. If you keep pressing <kbd>tab</kbd> (another 10+ times), you'll eventually get to the design panel, but you can only* get there with the top-most element on the canvas selected.

You cannot "select" an element and move to the design panel to adjust options for exactly this selected element.

*Well, you can also get to the design panel with the background element selected. You have to tab all the way around backwards - requires <kbd>shift</kbd> <kbd>tab</kbd> 70+ times to cycle through the entire WP menu.

**Shortcut to canvas and library:**
You _can_ always get to these, but it might require 30+ (<kbd>shift</kbd>) <kbd>tab</kbd> presses depending on where focus currently is. I'd classify this as _cumbersome_.

**Comment by pbakaus** [comment]
I'd say fixing this is definitely a requirement then.

One thought: We could consider not allowing you to tab through elements on the canvas without some other modifier key - that would allow you to always be able to tab to the design panel from currently selected element. Just a thought, either way seems workable.

**Comment by o-fernandez** [comment]
Sam is sorting out the shortcuts as part of the design work. #4243

**Comment by bmattb** [comment]
@samitron7 is blocked on this until she gets the confirmation from the Chrome team on these being okay to use.

**Comment by o-fernandez** [comment]
From @samitron7 over email. Recommendation for navigation between panels: 

MAC
CMD+Shift + < or >

PC
Ctrl+Shift + < or >

@samitron7 feel free to update! Otherwise, moving this to the sprint so it can be picked up.

**Comment by samitron7** [comment]
@omar you are so fast!!! Just wanted to point out that those are brackets and not arrow keys as **cmd**+ **shift** + **->** is reserved for text manipulation.

**Comment by barklund** [comment]
@o-fernandez @samitron7 that's unfortunately a very US-centric shortcut. This is the location of the angle brackets on a Danish Mac keyboard layout.
![Screenshot_20200903-175550~2](https://user-images.githubusercontent.com/637548/92177820-3dfafb00-ee0f-11ea-8f65-bd9661047eb9.png)

As you see, <kbd>shift</kbd> is the only difference between typing one or the other.

So Danish users will only ever be able to type one of those shortcuts.

And that's probably the case for other layouts as well.

**Comment by o-fernandez** [comment]
> @o-fernandez @samitron7 that's unfortunately a very US-centric shortcut. This is the location of the angle brackets on a Danish Mac keyboard layout.
> ![Screenshot_20200903-175550~2](https://user-images.githubusercontent.com/637548/92177820-3dfafb00-ee0f-11ea-8f65-bd9661047eb9.png)
> 
> As you see, shift is the only difference between typing one or the other.
> 
> So Danish users will only ever be able to type one of those shortcuts.
> 
> And that's probably the case for other layouts as well.

In that case, can we more simply define it as <kbd>CMD</kbd>+<kbd><</kbd> or <kbd>CMD</kbd>+<kbd>></kbd>?
If people need to use <kbd>SHIFT</kbd> as part of reaching the <kbd><</kbd> or <kbd>></kbd> isn't really important to define.

**Comment by barklund** [comment]
> > @o-fernandez @samitron7 that's unfortunately a very US-centric shortcut. This is the location of the angle brackets on a Danish Mac keyboard layout.
> > As you see, shift is the only difference between typing one or the other.
> > So Danish users will only ever be able to type one of those shortcuts.
> > And that's probably the case for other layouts as well.
> 
> In that case, can we more simply define it as CMD+< or CMD+>?
> If people need to use SHIFT as part of reaching the < or > isn't really important to define.

That works, yeah. Then people will have to figure out which other keys they might need to hold down to make those symbols, but that's pretty standard for such advanced shortcuts, I think.

**Comment by samitron7** [comment]
OMG...for real! So CMD+< goes to the Chrome browser settings page. I noticed that Figma is able to bypass that but most other browser based app will open up a the Chrome browser search settings page. I'm a little hesitant to hijack this shortcut. The other option was

cmd+option+ 1 or 2 or 3 for the 3 panels.

**Comment by barklund** [comment]
> OMG...for real! So CMD+< goes to the Chrome browser settings page.

Pressing <kbd>cmd</kbd> and then <kbd><</kbd> on US mac keyboard requires you to also hold the shift key, right? Otherwise it would be considered pressing <kbd>cmd</kbd>+<kbd>,</kbd>, no (which does open settings)?

Do you think it's confusing to just say <kbd>cmd</kbd>+<kbd><</kbd> if we actually mean <kbd>cmd</kbd>+<kbd>shift</kbd>+<kbd>,</kbd> (but only for US users, other users will not have `,` and `<` on the same key)?

**Comment by samitron7** [comment]
@barklund We need to say cmd+shift+< or >

**Comment by barklund** [comment]
Further conversation let us to instead use <kbd>Cmd</kbd><kbd>Option</kbd><kbd>1</kbd>/<kbd>2</kbd>/<kbd>3</kbd> for the three sections. Ticket to be updated.
